### DISCUSSION TITLE
TypeError: 'float' object is not subscriptable

### DISCUSSION THREAD

**Discussion by KrisTopherHao** [PaddlePaddle/PaddleOCR#12722] [Q&A]
请提供下述完整信息以便快速定位问题/Please provide the following information to quickly locate the problem

- 系统环境/System Environment：window10
- 版本号/Version：PaddleOCR-release-2.7    PPOCRLabel
- 运行指令/Command Code：python PPOCRLabel.py --lang ch
- 完整报错/Complete Error Message：
PS C:\Users\HEDY\Desktop\PaddleOCR-release-2.7\PPOCRLabel> python .\PPOCRLabel.py --lang ch
D:\Anaconda\Lib\site-packages\paramiko\transport.py:219: CryptographyDeprecationWarning: Blowfish has been deprecated
  "class": algorithms.Blowfish,
[2023/11/03 15:13:06] ppocr WARNING: When args.layout is false, args.ocr is automatically set to false
file name in openNext is  C:\Users\HEDY\Desktop\label\1.png
DirPath in importDirImages is C:/Users/HEDY/Desktop/label
unicodeFilePath is C:\Users\HEDY\Desktop\label\1.png
Using model from  paddle
'float' object is not subscriptable
Traceback (most recent call last):
  File "C:\Users\HEDY\Desktop\PaddleOCR-release-2.7\PPOCRLabel\libs\autoDialog.py", line 66, in run
    self.mainThread.saveFile(mode='Auto')
  File "C:\Users\HEDY\Desktop\PaddleOCR-release-2.7\PPOCRLabel\PPOCRLabel.py", line 1894, in saveFile
    self._saveFile(imgidx, mode=mode)
  File "C:\Users\HEDY\Desktop\PaddleOCR-release-2.7\PPOCRLabel\PPOCRLabel.py", line 1945, in _saveFile
    if annotationFilePath and self.saveLabels(annotationFilePath, mode=mode):
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\HEDY\Desktop\PaddleOCR-release-2.7\PPOCRLabel\PPOCRLabel.py", line 1230, in saveLabels
    trans_dic = {"label": box[1][0], "points": box[0], "difficult": False}
                          ~~~~~~^^^
**TypeError: 'float' object is not subscriptable**


sourcecode：
 for box in self.result_dic:
            **trans_dic = {"label": box[1][0], "points": box[0], "difficult": False}**   //*********
            if self.kie_mode:
                if len(box) == 3:
                    trans_dic.update({"key_cls": box[2]})
                else:
                    trans_dic.update({"key_cls": "None"})
            if trans_dic["label"] == "" and mode == 'Auto':
                continue
            shapes.append(trans_dic)

**Comment by NishantRoy19** [comment]
Tried the code below:
**env:** 
m1 macbook air with sonoma version 14.4
conda with Python 3.10.13 with

Package                                  Version
Flask                                        ==3.0.2
fuzzywuzzy                             ==0.18.0
h11                                            ==0.14.0
h5py                                         ==3.10.0
keras                                        ==3.0.5
numpy                                      ==1.23.5
opencv-contrib-python         ==4.9.0.80
opencv-python                       ==4.9.0.80
paddleocr                                ==2.6.0.1
paddlepaddle                          ==2.6.0
pandas                                     ==2.2.1
pdf2image                               ==1.17.0
pillow                                       ==10.2.0
pip                                           ==23.3.1
PyMuPDF                                ==1.21.1
pyparsing                                ==3.1.2
scikit-image                            ==0.22.0
scikit-learn                              ==1.4.1.post1
scipy                                        ==1.12.0
tensorflow                               ==2.16.1
torch                                        ==2.2.1
torchvision                              ==0.17.1
ultralytics                                ==8.1.29

**code:**
from paddleocr import PaddleOCR,draw_ocr
ocr = PaddleOCR(use_angle_cls=True, lang='en')
img_path = 'PaddleOCR/doc/imgs_en/img_12.jpg'
result = ocr.ocr(img_path, cls=True)
for idx in range(len(result)):
    res = result[idx]
    for line in res:
        print(line)

from PIL import Image
result = result[0]
image = Image.open(img_path).convert('RGB')
boxes = [line[0] for line in result]
txts = [line[1][0] for line in result]
scores = [line[1][1] for line in result]
im_show = draw_ocr(image, boxes, txts, scores, font_path='/path/to/PaddleOCR/doc/fonts/simfang.ttf')
im_show = Image.fromarray(im_show)
im_show.save('result.jpg')

**giving the error:**
TypeError: 'float' object is not subscriptable

**Comment by SWHL** [comment]
Since PPOCRLabel has moved, please go to [PPOCRLabel](https://github.com/PFCCLab/PPOCRLabel) to ask questions if you have any questions!
This issue has been closed first.
